File: Data Structures & Algorithms/partition-equal-subset-sum/submission-1.java
class Solution {
    public boolean canPartition(int[] nums) {
        int sum=0;
        for(int i=0;i<nums.length;i++)sum+=nums[i];
        if(sum%2!=0)return false;
        int required = sum/2;
        boolean[] dp=new boolean[required+1];
        dp[0]=true;
        for(int i=0;i<nums.length;i++){
            for(int j=required;j>=nums[i];j--){
                dp[j]=dp[j]||dp[j-nums[i]];
                if(dp[required])return true;
            }
        }
        return dp[required];
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = defaultdict(list)
        for num in nums:
            freq[num]=freq.get(num,0)+1
        sortedByValues=sorted(freq.items(),key=lambda x:x[1], reverse=True)
        ans =[0]*k
        for i in range(k):
            ans[i]=sortedByValues[i][0]
        return ans
